### ISSUE TITLE
apiserver: componentstatuses probes doesn't work correctly

### ISSUE THREAD

**Issue by kayrus** [kubernetes/kubernetes#35773]
In case when etcd cluster is configured with the tls auth `kubectl get cs` returns errors:

``` sh
$ kubectl get componentstatuses
NAME                 STATUS      MESSAGE                                                               ERROR
scheduler            Healthy     ok                                                                    
controller-manager   Healthy     ok                                                                    
etcd-0               Unhealthy   Get https://172.16.4.205:2379/health: remote error: bad certificate   
etcd-1               Unhealthy   Get https://172.16.4.54:2379/health: remote error: bad certificate    
etcd-2               Unhealthy   Get https://172.16.4.206:2379/health: remote error: bad certificate
```

In case when scheduler and controller-manager are configured to listen on private interface (nigher  wildcard nor localhost) `kubectl get cs` also returns errors:

``` sh
$ kubectl get componentstatuses
NAME                 STATUS      MESSAGE                                                                                        ERROR
scheduler            Unhealthy   Get http://127.0.0.1:10251/healthz: dial tcp 127.0.0.1:10251: getsockopt: connection refused
controller-manager   Unhealthy   Get http://127.0.0.1:10252/healthz: dial tcp 127.0.0.1:10252: getsockopt: connection refused
```

/cc @Bregor

The draft fix is [here](https://github.com/kayrus/kubernetes/commit/aa7c91c0b6f39503b98f127b42274e35256f2b91)

**Comment by Bregor** [comment]
Example manifests for `scheduler` and `controller-manager`:

controller-manager: https://gist.github.com/Bregor/27eb921f1a8f580f3dd5c3fbb234aefd#file-controller-manager-yaml

scheduler: https://gist.github.com/Bregor/27eb921f1a8f580f3dd5c3fbb234aefd#file-scheduler-yaml

Result of `kubectl describe po -n kube-system kube-scheduler-10.83.8.200`: https://gist.github.com/Bregor/27eb921f1a8f580f3dd5c3fbb234aefd#file-kubectl-describe-po-n-kube-system-kube-scheduler-10-83-8-200

Result of `kubectl describe po -n kube-system kube-controller-manager-10.83.8.200`: https://gist.github.com/Bregor/27eb921f1a8f580f3dd5c3fbb234aefd#file-kubectl-describe-po-n-kube-system-kube-controller-manager-10-83-8-200

Pay attention to following strings in `describe`:

    Liveness:		http-get http://:10252/healthz delay=15s timeout=1s period=10s #success=1 #failure=3
    Readiness:		http-get http://:10252/healthz delay=15s timeout=1s period=10s #success=1 #failure=3
    ...
    Conditions:
      Type		Status
      Initialized 	True
      Ready 	True
      PodScheduled 	True

Especially to `http-get http://:10252/healthz`

Despite of above there is log of kubelet:

    Oct 28 12:29:08 web01-amplifr-ru kubelet[26193]: I1028 12:29:08.122498   26193 prober.go:161] HTTP-Probe Host: http://10.83.8.200, Port: 10251, Path: /healthz
    Oct 28 12:29:08 web01-amplifr-ru kubelet[26193]: I1028 12:29:08.122585   26193 prober.go:161] HTTP-Probe Host: http://10.83.8.200, Port: 10251, Path: /healthz

As you can see, in reality, there are checks done right in kubelets.

And finally:

    $ kubectl get cs
    NAME                 STATUS      MESSAGE                                                                                        ERROR
    controller-manager   Unhealthy   Get http://127.0.0.1:10252/healthz: dial tcp 127.0.0.1:10252: getsockopt: connection refused
    scheduler            Unhealthy   Get http://127.0.0.1:10251/healthz: dial tcp 127.0.0.1:10251: getsockopt: connection refused
    etcd-1               Healthy     {"health": "true"}
    etcd-0               Healthy     {"health": "true"}
    etcd-2               Healthy     {"health": "true"}

There is no any real checks for `http://127.0.0.1:10251/healthz`

**Comment by elcct** [comment]
@kayrus  Looks like your commit could fix the issue reported here:

https://github.com/kubernetes/kubernetes/issues/27343

At the moment when checking for etcd health, certificates are not being used so when etcd has enabled `-client-cert-auth` etcd cluster is falsely reported as unhealthy.

**Comment by caio1982** [comment]
Does the draft fix really fixes #27343? I have seen this exact behavior today and I have been wondering about it. Is any of you still actively working on this draft, just out of curiosity?

**Comment by justaugustus** [comment]
We're planning on deprecating this command.
Watch this Feature thread for updates: https://github.com/kubernetes/features/issues/553

/close
